tsdb/engine/wal/wal.go

```diff
@@ -0,0 +1,1591 @@
+	for _, fn := range metaFiles {
+		a, err := l.readMetadataFile(fn)
+		if err != nil {
+			return err
+		}
+
+		// loop through the seriesAndFields and add them to the index and the collection to be written to the index
+		for _, sf := range a {
+			for k, mf := range sf.Fields {
+				measurementFieldsToSave[k] = mf
+
+				m := index.CreateMeasurementIndexIfNotExists(string(k))
+				for name, _ := range mf.Fields {
+					m.SetFieldName(name)
+				}
+				mf.Codec = tsdb.NewFieldCodec(mf.Fields)
+				measurementFields[m.Name] = mf
+			}
+
+			for _, sc := range sf.Series {
+				seriesToCreate = append(seriesToCreate, sc)
+
+				sc.Series.InitializeShards()
+				index.CreateSeriesIndexIfNotExists(tsdb.MeasurementFromSeriesKey(string(sc.Series.Key)), sc.Series)
+			}
+		}
+	}
+
+	if err := l.Index.WriteIndex(nil, measurementFieldsToSave, seriesToCreate); err != nil {
+		return err
+	}
+
+	// now remove all the old metafiles
+	for _, fn := range metaFiles {
+		if err := os.Remove(fn); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// DeleteSeries will flush the metadata that is in the WAL to the index and remove
+// all series specified from the cache and the segment files in each partition. This
+// will block all writes while a compaction is done against all partitions. This function
+// is meant to be called by bz1 BEFORE it updates its own index, since the metadata
+// is flushed here first.
+func (l *Log) DeleteSeries(keys []string) error {
+	// we want to stop any writes from happening to ensure the data gets cleared
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	if err := l.flushMetadata(); err != nil {
+		return err
+	}
+
+	for _, p := range l.partitions {
+		p.deleteSeries(keys)
+	}
+
+	return nil
+}
+
+// readMetadataFile will read the entire contents of the meta file and return a slice of the
+// seriesAndFields objects that were written in. It ignores file errors since those can't be
+// recovered.
+func (l *Log) readMetadataFile(fileName string) ([]*seriesAndFields, error) {
+	f, err := os.OpenFile(fileName, os.O_RDWR, 0666)
+	if err != nil {
+		return nil, err
+	}
+
+	a := make([]*seriesAndFields, 0)
+
+	length := make([]byte, 8)
+	for {
+		// get the length of the compressed seriesAndFields blob
+		_, err := f.Read(length)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			f.Close()
+			return nil, err
+		}
+
+		dataLength := btou64(length)
+		if dataLength == 0 {
+			break
+		}
+
+		// read in the compressed block and decod it
+		b := make([]byte, dataLength)
+
+		_, err = f.Read(b)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			// print the error and move on since we can't recover the file
+			l.logger.Println("error reading lenght of metadata: ", err.Error())
+			break
+		}
+
+		buf, err := snappy.Decode(nil, b)
+		if err != nil {
+			// print the error and move on since we can't recover the file
+			l.logger.Println("error reading compressed metadata info: ", err.Error())
```

Comment:
Minor, but it looks like the string has a trailing space. This isn't necessary since `Println` concatenates using a space. This occurs in a couple of other places.